app/assets/src/components/views/compare/SamplesHeatmapView/SamplesHeatmapView.jsx

```diff
@@ -456,6 +460,64 @@ class SamplesHeatmapView extends React.Component {
     };
   }
 
+  fetchBackgroundData() {
+    if (this.lastRequestToken)
+      this.lastRequestToken.cancel("Parameters changed");
+    this.lastRequestToken = axios.CancelToken.source();
+
+    return updateHeatmapBackground(
+      {
+        sampleIds: this.state.sampleIds,
+        taxonIds: Array.from(this.state.allTaxonIds),
+        removedTaxonIds: Array.from(this.removedTaxonIds),
+        background: this.state.selectedOptions.background,
+        heatmapTs: this.props.heatmapTs,
+      },
+      this.lastRequestToken.token
+    );
+  }
+
+  async fetchBackground() {
+    this.setState({ loading: true });
+    let heatmapData = await this.fetchBackgroundData();
```

Comment:
Should we call the data backgroundData? Or is it `fetchBackgroundData` that is deceiving?

Reply:
Oops, renamed to `backgroundData`. `heatmapData` is carried over from an old variable I forgot to rename.